packages/admin/cms-admin/src/blocks/InternalLinkBlock.tsx

```diff
@@ -71,6 +73,30 @@ export const InternalLinkBlock: BlockInterface<InternalLinkBlockData, State, Int
         return true;
     },
 
+    dependencies: (state) => {
+        const dependencies: BlockDependency[] = [];
+
+        if (state.targetPage?.id) {
+            dependencies.push({
+                targetGraphqlObjectType: "PageTreeNode",
+                id: state.targetPage.id,
+            });
+        }
+
+        return dependencies;
+    },
+
+    createCopy: (state, { idsMap }) => {
```

Comment:
This should work as the document and use the block's input as argument and provide the block's oupt